File: 28.py
#!/usr/bin/env python

"""This script solves the Project Euler problem "Number spiral diagonals". The
problem is: What is the sum of the numbers on the diagonals in a 1001 by 1001
spiral formed in the same way?
"""

from __future__ import division
import argparse


def main(args):
    """Number spiral diagonals"""

    sum = 1  # Middle number
    current_width = 1
    increment = 0
    number = 1

    while current_width < args.width:
        current_width += 2
        increment += 2
        for _ in range(4):
            number += increment
            sum += number

    print(sum)

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Number spiral diagonals')
    parser.add_argument(
        'width', metavar='WIDTH', type=int, default=1001, nargs='?',
        help='The width of the number spiral')
    args = parser.parse_args()

    main(args)

File: 28.js
#!/usr/bin/env node

// Number spiral diagonals

var program = require('commander');

program
    .version('0.1.0')
    .description('Number spiral diagonals')
    .option('-w, --width <int>', 'The width of the number spiral', Number, 1001)
    .parse(process.argv);

var sum = 1;
var current_width = 1;
var increment = 0;
var number = 1;

while (current_width < program.width) {
    current_width += 2;
    increment += 2;
    for (var i = 0; i < 4; i++) {
        number += increment;
        sum += number;
    }
}

console.log(sum);
